File: week_1/03_01_find_alphabet_occurrence.py
def find_max_occurred_alphabet(string):
    alphabet_array = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s",
                      "t", "u", "v", "x", "y", "z"]
    
    max_occurrence=0 #최빈 변수
    max_alphabet=alphabet_array[0] #가장 많이 나온 알파벳

    for alphabet in alphabet_array:
        occurrence=0 #각 알파벳 배열이 바뀔때 마다 0으로 초기화
        for char in string:
            if alphabet == char:
                occurrence += 1
        
        if max_occurrence < occurrence: #최대 빈도수 보다 큰 빈도수가 생기면
            max_occurrence=occurrence 
            max_alphabet=alphabet
    return max_alphabet

result = find_max_occurred_alphabet
print("정답 = a 현재 풀이 값 =", result("Hello my name is sparta"))
print("정답 = a 현재 풀이 값 =", result("Sparta coding club"))
print("정답 = s 현재 풀이 값 =", result("best of best sparta"))

File: week_1/03_01_find_alphabet_occurrence.cpp
#include <iostream>
#include <string>
using namespace std;

char alphabet_array[26] = { 'a', 'b', 'c', 'd', 'e', 'f','g' , 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's',
't', 'u', 'v', 'x', 'y', 'z'}; // 알파벳 원소를 담는 배열

char find_max_occurred_alphabe(string str) {
	int max_occurrence = 0; //최빈 변수
	char max_alphabet = alphabet_array[0]; //최빈 알파벳 

	for (int i = 0; i < 26; i++) {
		int occurrence = 0; //빈도수 체크 변수
		for (int j = 0; j < str.length(); j++) { //입력받은 str의 알파벳 빈도수 체크
			if (alphabet_array[i] == str[j]) 
				occurrence += 1;
		}
		if (max_occurrence < occurrence) { //만약 기존 MAX빈도수보다 for문을 막 돌린 빈도수가 크다면
			// 교체
			max_occurrence = occurrence; 
			max_alphabet = alphabet_array[i];
		}
	}
	return max_alphabet;
}

int main(void) {
	char answer = find_max_occurred_alphabe("Hello my name is sparta");
	cout << answer << '\n';
	answer= find_max_occurred_alphabe("Sparta coding club");
	cout << answer << '\n';
	answer = find_max_occurred_alphabe("best of best sparta");
	cout << answer << '\n';
	return 0;
}
